cpp/src/arrow/compute/kernels/cast.h

```diff
@@ -35,10 +35,23 @@ class DataType;
 namespace compute {
 
 struct ARROW_EXPORT CastOptions {
-  CastOptions() : allow_int_overflow(false), allow_time_truncate(false) {}
+  CastOptions()
+      : allow_int_overflow(false),
+        allow_time_truncate(false),
+        allow_float_truncate(true) {}
+
+  explicit CastOptions(bool safe)
```

Comment:
Perhaps annotate these methods as internal / experimental. I'm not sure what will be the best way longer-term to manage these flags